src/Components/InfiniteScroll.tsx

```diff
@@ -9,80 +9,64 @@ import {
   lifecycle,
   pure,
   setDisplayName,
-  withHandlers,
-  withProps,
   withPropsOnChange,
-  withState,
   withStateHandlers
 } from "recompose";
 
-import { debounce, throttle } from "lodash";
+import { isEmpty } from "lodash";
 
 import { Query } from "../Query";
 import { withModelQuery } from "../WithModelQuery";
 import { average, calculateGroupHeight, omitProps } from "../Helpers";
+import { IWithDelayedHandlers, withDelayedHandlers } from "../WithDelayedHandlers";
+import { withLoadingIndicator } from "../WithLoadingIndicator";
+import { InfiniteScrollPreviousPage } from "./InfiniteScrollPreviousPage";
 
-export interface IInfiniteScrollProps<T extends IModelData> {
+export interface IInfiniteScrollProps<T extends IModelData> extends IWithDelayedHandlers {
   modelName: string;
   query: IQueryParams | IQueryBuilder;
   containerComponent: React.ComponentType<{ onScroll: React.UIEventHandler } & any>;
   modelComponent: React.ComponentType<{ model: T } & any>;
   modelComponentProps?: any;
-  debounceTime?: number;
   disableVirtualScrolling?: boolean;
-  contentPlaceHolderComponent?: React.ComponentType<{ height: number }>;
+  contentPlaceHolderComponent?: React.ComponentType<{ height: number | string }>;
+  contentPlaceHolderComponentProps?: any;
 
   /* Any extra props will be passed-through to the container component */
   [key: string]: any;
 }
 
-export interface IInfiniteScrollHeightMapProps {
-  pageHeightMap: { [key: string]: number };
-  estimatedPageHeight: number;
-}
-
-export interface IInfiniteScrollInternalProps<T extends IModel<IModelData>> extends Omit<IInfiniteScrollProps<any>, "query">, IInfiniteScrollHeightMapProps {
-  query: IQueryManager<T>;
-  updatePageHeightMap: (query: IQueryManager<any>) => any;
-  updateQuery: (queryBuilder: IQueryBuilder) => void;
+export interface IInfiniteScrollStateProps<T extends IModelData> {
+  currentItems: T[];
+  currentPage: number;
+  currentPageStartMarkerRef: React.RefObject<any>;
+  currentPageEndMarkerRef: React.RefObject<any>;
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
   lastScrollTop: number;
-  nextPageHeightRef: React.RefObject<any>;
-  handleScrollDebounced: (e) => void;
-  handleScrollThrottled: (e) => void;
-  handleScrollPersistingEvent: (e) => void;
-  previousPlaceHolderHeight: number;
+  nextPage: number;
+  nextPageQuery: IQueryBuilder;
```

Comment:
These are out of order.  Not sure it's really critical because it's nice that the markers are next to each other.

Reply:
fixed